templates/ts/client-node.ts

```diff
@@ -0,0 +1,135 @@
+// tslint:disable
+import fetch from 'node-fetch';
+import { RequestInit } from 'node-fetch';
+import AbortController from 'abort-controller';
+import { createReturnTypeValidator, ClassValidator, ValidationError } from './common';
+import {
+  schema,
+  InternalServerError,
+  {{#exceptions}}
+  {{name}},
+  {{/exceptions}}
+  {{#classes}}
+  {{name}},
+  {{/classes}}
+  {{#enums}}
+  {{name}},
+  {{/enums}}
+  {{#bypassTypes}}
+  {{name}},
+  {{/bypassTypes}}
+} from './interfaces';
+
+export class TimeoutError extends Error {
+}
+
+export {
+  ValidationError,
+};
+
+export interface Options extends Pick<RequestInit, 'agent' | 'redirect' | 'follow' | 'compress'> {
+  fetchImplementation?: typeof fetch;
+  timeoutMs?: number;
+  headers?: Record<string, string>;
+}
+
+{{#clientContext}}
+export type Context = ClientContext;
+{{/clientContext}}
+
+{{#classes}}
+{{^attributes}}
+export interface {{name}} {
+  {{#methods}}
+  {{name}}({{#clientContext}}ctx: Context ,{{/clientContext}}{{#parameters}}{{name}}{{#optional}}?{{/optional}}: {{{type}}}{{^last}}, {{/last}}{{/parameters}}): Promise<{{{returnType}}}>;
+  {{/methods}}
+}
+
+export class {{name}}Client {
+  public static readonly methods = [
+    {{#methods}}
+    '{{name}}',
+    {{/methods}}
+  ];
+  public static readonly validators: ClassValidator = createReturnTypeValidator(schema, '{{{name}}}');
+
+  protected readonly props = schema.definitions.{{{name}}}.properties;
+
+  public readonly validators: ClassValidator; // We don't have class name in method scope because mustache sux
+
+  public constructor(protected readonly serverUrl: string, protected readonly options: Options = {}) {
+    this.validators = {{{name}}}Client.validators;
+  }
+  {{#methods}}
+
+  public async {{name}}({{#clientContext}}ctx: Context ,{{/clientContext}}{{#parameters}}{{name}}{{#optional}}?{{/optional}}: {{{type}}}, {{/parameters}}options?: Options): Promise<{{{returnType}}}> {
+    const body = {
+      args: {
+        {{#parameters}}
+        {{name}},
+        {{/parameters}}
+      },
+      {{#clientContext}}
+      context: ctx,
+      {{/clientContext}}
+    };
+
+    const mergedOptions = {
+      ...this.options,
+      ...options,
+    };
+
+    if (mergedOptions.timeoutMs) {
+      const controller = new AbortController();
+      const timeout = setTimeout(() => controller.abort(), mergedOptions.timeoutMs);
+      (mergedOptions as any).signal = controller.signal;
+      delete mergedOptions.timeoutMs;
+    }
+
+    try {
+      const response = await (mergedOptions.fetchImplementation || fetch)(`${this.serverUrl}/{{name}}`, {
```

Comment:
@vogre was this what you were looking for?

Reply:
Separation of `client-browser` and `client-node` got me there.